.github/workflows/linux_hip.yml

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2021 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+name: Linux HIP CI (Debug)
+
+on: [pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container: stellargroup/build_env:hip
```

Comment:
Where is this image built? We might want to version it similarly to the main image, and probably base this on the main image.

Reply:
This image is built on top of stellargroup/build_env:2 it's on the [add_hip_docker_image_for_github_actions branch](https://github.com/STEllAR-GROUP/docker_build_env/tree/add_hip_docker_image_for_github_actions) :)

Reply:
So do you prefer me to put the image on master but on another directory like `hip_build_env`? I thought about it but it required more changes to CI so I opted for just a separate branch

Reply:
Absolutely!

On the other hand this isn't as important now that the builds are working again on rostam, but I suppose it doesn't hurt to have a more reliable environmet on gh actions even if the tests aren't run.

Reply:
So should I close this PR?

Reply:
So now the hip docker image is in another directory on the master branch, it should be ready to go once the CI ran :)

Reply:
I guess the image tag should be updated here?